codex-cli/src/approvals.ts

```diff
@@ -115,6 +115,11 @@ export function canAutoApprove(
     try {
       bashCmd = parse(command[2], env);
     } catch (e) {
+      // Log error for debugging
+      if (env.NODE_ENV === "development") {
+        // eslint-disable-next-line no-console
+        console.error("Shell parse error in canAutoApprove:", e);
+      }
```

Comment:
There's a logger you can import already in the codebase, which will log to a file when in env var `DEBUG=true`

```suggestion
      if (isLoggingEnabled()) {
        log(`Bash command parsing failed for: ${command[2]}\nCaused by: ${e instanceof Error ? e.message : String(e)}`);
      }
```

Reply:
yeah it is 
> @[tibo-openai](https://github.com/tibo-openai) tibo-openai **changed the title** Add Dev-Mode Error Logging for Shell Parse Failures feat: add dev-mode error logging for shell parse failures [7 hours ago](https://github.com/openai/codex/pull/271#event-17312577929)